kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/OperatorChangeDetectionST.java

```diff
@@ -147,8 +141,40 @@ void shouldRolloutDeploymentWhenFilterConfigurationChanges(String namespace) {
         LOGGER.info("Kafka proxy filter updated");
 
         // Then
+        assertDeploymentUpdated(namespace, kubeClient, originalChecksum);
+    }
+
+    @Test
+    void shouldUpdateDeploymentWhenKafkaProxyChanges(String namespace) {
+        // Given
+        KubeClient kubeClient = kubeClient(namespace);
+        kroxylicious.deployPortIdentifiesNodeWithNoFilters(kafkaClusterName);
+
+        String originalChecksum = getInitialChecksum(namespace, kubeClient);
+
+        KafkaProxy kafkaProxy = kubeClient.getClient().resources(KafkaProxy.class).inNamespace(namespace)
+                .withName(Constants.KROXYLICIOUS_PROXY_SIMPLE_NAME).get();
+
+        // @formatter:off
+        KafkaProxyBuilder updatedKafkaProxy = kafkaProxy.edit()
+                .editOrNewSpec()
+                .withPodTemplate(new PodTemplateSpecBuilder().build())
+                .endSpec();
+
+        // @formatter:on
+
+        // When
+        resourceManager.createOrUpdateResourceWithWait(updatedKafkaProxy);
+        LOGGER.info("Kafka proxy updated");
+
+        // Then
+        assertDeploymentUpdated(namespace, kubeClient, originalChecksum);
+    }
+
+    private static void assertDeploymentUpdated(String namespace, KubeClient kubeClient, String originalChecksum) {
         await().atMost(Duration.ofSeconds(90)).untilAsserted(() -> {
-            Deployment proxyDeployment = kubeClient.getDeployment(namespace, "simple");
+            Deployment proxyDeployment = kubeClient.getDeployment(namespace, Constants.KROXYLICIOUS_PROXY_SIMPLE_NAME);
+            Assertions.assertThat(proxyDeployment).isNotNull();
```

Comment:
The NPE triggered locally and demonstrated I don't understand the docs as only assertion errors are retried...

I reproduced the failure twice but the system tests auto cleaned up the cluster and I didn't find anything in the cluster dump to explain it. 

FYI: @franvila

Reply:
Yeap, it is hard to investigate as I had to copy the `logCollector` code right after the `update` to try to get all the info as possible from the pods that are terminated. I uploaded some logs to the issue #2150. Somehow there is some entity that kill the pod, but I don't know who and why